### ISSUE TITLE
Add a filter+cache mechanism for addon needing to be called

### ISSUE THREAD

**Issue by dvojtise** [eclipse/gemoc-studio-modeldebugging#112]
in relation with #111,
The computation of the order may take some time and decrease execution time since it is currently done before and after each step.

We should provide a way to cache this for a given execution.

Additionally, we can add in the addon definition some information about the list of notification/event that the addon is really interested in.

By default, an addon would notified about everything, but if we can filter on which events an addon must be notified, then we would also save some time by not calling the "empty" method for it.

see  how the addons are called in https://github.com/eclipse/gemoc-studio-modeldebugging/blob/master/framework/execution_framework/plugins/org.eclipse.gemoc.executionframework.engine/src/org/eclipse/gemoc/executionframework/engine/core/AbstractExecutionEngine.java

**Comment by ebousse** [comment]
This is interesting, but before making such a change it would be nice to measure how much could be gained. Is this really a big waste of time, to call methods that do nothing? 

It seems the JVM takes care of it automatically, since it inlines methods called regularly, which in our case means deleting the method call since the replacement is an empty method body. See https://stackoverflow.com/questions/43482380/does-calling-an-empty-method-in-java-use-resources

**Comment by dvojtise** [comment]
Agreed. This is why I put it in a separate issue. The gain will probably be more on the add-on introspection loop and priority computation sorting. -------- Message original --------Objet : Re: [eclipse/gemoc-studio-modeldebugging] Add a filter+cache mechanism for addon needing to be called (#112)De : Erwan Bousse À : eclipse/gemoc-studio-modeldebugging Cc : Didier Vojtisek ,Author This is interesting, but before making such a change it would be nice to measure how much could be gained. Is this really a big waste of time, to call methods that do nothing?
It seems the JVM takes care of it automatically, since it inlines methods called regularly, which in our case means deleting the method call since the replacement is an empty method body. See https://stackoverflow.com/questions/43482380/does-calling-an-empty-method-in-java-use-resources
